core/src/test/scala/org/apache/spark/status/protobuf/KVStoreProtobufSerializerSuite.scala

```diff
@@ -176,4 +177,83 @@ class KVStoreProtobufSerializerSuite extends SparkFunSuite {
     assert(result.stageId == input.stageId)
     assert(result.stageAttemptId == input.stageAttemptId)
   }
+
+  test("Application Environment Info") {
+    val input = new ApplicationEnvironmentInfoWrapper(
+      new ApplicationEnvironmentInfo(
+        runtime = new RuntimeInfo(
+          javaVersion = "1.8",
+          javaHome = "/tmp/java",
+          scalaVersion = "2.13"),
+        sparkProperties = Seq(("1", "2")),
+        hadoopProperties = Seq(("1", "2")),
```

Comment:
I would suggest setting different values for these fields in case of mistakes on the fields of serializer and deserializer